examples/tests/invalid/class.test.w

```diff
@@ -163,4 +163,55 @@ inflight class Jet extends Plane{
 }
 
 new std.Resource();
-//^^^^^^^^^^^^^^^^ Cannot instantiate abstract class "Resource"
\ No newline at end of file
+//^^^^^^^^^^^^^^^^ Cannot instantiate abstract class "Resource"
+
+// Class with field and method of the same name (field comes first)
+class C12 {
+  z: num;
+  z(): num {
+  //^ Symbol "z" is already defined
+    // The method body should still be type checked
+    2 + "2";
+    //^ Expected type to be "num", but got "str" instead
+    this.z == 5; // OK
```

Comment:
Within the body of `z() {...}`, `this.z` is typed as a `num` and not as a function. Is this correct? This can probably be debated but I think either behavior is OK since the user has a naming issue they need to fix

Reply:
Yeah, I think it's ok either way. Let's give it a trt as you implemented and see how it goes